polling/assets.go

```diff
@@ -298,6 +303,65 @@ func (d *DataSources) PullAllPEGAssets(oprversion uint8) (pa PegAssets, err erro
 	return pa, nil
 }
 
+// PullBestPrice pulls the best asset price we can find for a given asset.
+// Params:
+//		asset		Asset to pull pricing data
+//		reference	Time reference to determine 'staleness' from
+//		sources		Map of datasources to pull the price quote from.
+func (d *DataSources) PullBestPrice(asset string, reference time.Time, sources map[string]IDataSource) (pa PegItem, err error) {
```

Comment:
the way this works at the moment is:
1. it goes through the apis by the priority order
2. if it encounters a result that is not stale, it will return that
3. if all responses are stale, it returns the most recent one

1 and 2 work (albeit with some changes below), but i had a question for 3: if all api sources are stale, should we give them the newest, or abide by the intended configuration? you could make the argument that if all sources are stale there's likely something wrong (ie markets closed unexpectedly) and we should serve them with the stale data from the api they configured

Reply:
So if all sources are stale, return the highest priority one? I can make that change, that does seem reasonable.